### PR TITLE
Testestestest

### PR DESCRIPTION
**Pull request by bwan2**
Completed:

- Finished all the unit tests(based off mocha/chai) for the current system
- Refactored to accommodate tests
- Created two separate docker environments for dev/test
- tests are run using boundary cases generated from ACTS.

Run:

To test this, run `npm test`. This will create a test environment and run all the tests there. There should be 110 tests passing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,20 +13,26 @@ Pinocchio server in node.js. This server will handle backend requests and serve
 
 The development environment is using a dynamoDB local instance. 
 
-1. Configure `config.js` for the node server to use. **Note:** `mailAuth` requires using less secure access for gmail accounts (https://myaccount.google.com/lesssecureapps). Please ask Brian for email credentials.
-2. Set up `aws-cli` with teh following configuration:
+1. Configure `config.js` for the node server to use. Be sure to point dynamodb to the correct endpoint and set to use `dev`. **Note:** `mailAuth` requires using less secure access for gmail accounts (https://myaccount.google.com/lesssecureapps). Please ask Brian for email credentials.
```
Comment (miajhaochen) [inline README.md:16]: README is super greattttt! 
However, I actually have applied for an email `pinocchio.data61@gmail.com` to send the email. So I am wondering what the email credentials you have?

File: run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#export DEBUG="true"
+
+# start test instance
+docker-compose -f docker-test.yml up -d --build
+
+# create/delete tables in test db
+npm run create-tables
+
+# load test data
+node src/tests/testData/loadData.js
```
Comment (miajhaochen) [inline run_tests.sh:14]: why not `npm run load-test-data`

File: src/tests/accountsTest.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+              // current test request body
+              if(DEBUG) console.log(csvRow);
+              var status = parseInt(csvRow.status);
+
+              // send the request 
+              chai.request(app)
+              .post('/api/accounts/register')
+              .set('content-type', 'application/x-www-form-urlencoded')
+              .type('form')
+              .send('firstname=' + csvRow.firstname)
```
Comment (miajhaochen) [inline src/tests/accountsTest.js]: you may use "`firstname=${csvRow.firstname}`" ,  just the syntax

File: src/tests/accountsTest.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+  // loads register-unit.csv that has combinations of valid/invalid tests
+  // this csv file was created using ACTS.
+  function accountRegistrationTest(){
+    fs.readFile('./src/tests/testData/csvs/register-unit.csv', 'UTF-8', function(err, csv){
+      if(err) console.log(err);
+      jquerycsv.toObjects(csv, {}, function (err, csvData) {
+        if (err) { console.log(err); }
+        // async to get each csvRow and reuse the describe/it
+        async.each(csvData, function(csvRow, callback){
+          describe("POST /api/accounts/register", () =>{
```
Comment (miajhaochen) [inline src/tests/accountsTest.js]: do we have to consider invalid cases? for example, the email is incorrect format

**Comment by miajhaochen** [comment]
I have solved the conflict for you. However, there is still some errors from ESlint looool
For example, do we need to use swagger to generate document for test code? 
and naming convention error generated from `camelcase` ^____^

image

File: src/tests/accountsTest.js (hunk 4 above)
Comment (miajhaochen) [reply src/tests/accountsTest.js]: If it's too much, we can skip looool

**Comment by miajhaochen** [comment]
after I ran `npm test`, it shows this error: 

image

but it does run `npm install`, not sure why QQ
image
